### DISCUSSION TITLE
How can I use poetry with PEP-621 style `pyproject.toml`?

### DISCUSSION THREAD

**Discussion by kunaltyagi** [python-poetry/poetry#5833] [Q&A] [answered]
[PEP-621](https://peps.python.org/pep-0621/) shows the following `pyproject.toml` file (keeping only relevant sections for this discussion):
```toml
[project]
# ...
requires-python = ">=3.8"
# ...

dependencies = [
  "httpx",
  "gidgethub[httpx]>4.0.0",
  "django>2.1; os_name != 'nt'",
  "django>2.0; os_name == 'nt'"
]

[project.optional-dependencies]
test = [
  "pytest < 5.0.0",
  "pytest-cov[all]"
]
```

Does poetry support dependencies in this format? I tried a naive `poetry install --no-root` and got an error regarding `missing [tool.poetry]` section (which to be fair, I don't have)

**Comment by branchv** [comment, marked as answer]
Not currently. The next major version (2.0) will support PEP 621, you can follow the progress at https://github.com/python-poetry/roadmap/issues/3
